lib/streamlit/elements/beta_altair.py

```diff
@@ -0,0 +1,312 @@
+            Data to be plotted.
+
+        width : int
+            The chart width in pixels. If 0, selects the width automatically.
+
+        height : int
+            The chart height in pixels. If 0, selects the height automatically.
+
+        use_container_width : bool
+            If True, set the chart width to the column width. This takes
+            precedence over the width argument.
+
+        Example
+        -------
+        >>> chart_data = pd.DataFrame(
+        ...     np.random.randn(50, 3),
+        ...     columns=["a", "b", "c"])
+        ...
+        >>> st.beta_bar_chart(chart_data)
+
+        """
+        proto = BetaVegaLiteChartProto()
+        chart = generate_chart("bar", data, width, height)
+        marshall(proto, chart, use_container_width)
+        last_index = last_index_for_melted_dataframes(data)
+
+        return self.dg._enqueue("beta_bar_chart", proto, last_index=last_index)
+
+    def beta_altair_chart(self, altair_chart, use_container_width=False):
+        """Display a chart using the Altair library.
+
+        Parameters
+        ----------
+        altair_chart : altair.vegalite.v2.api.Chart
+            The Altair chart object to display.
+
+        use_container_width : bool
+            If True, set the chart width to the column width. This takes
+            precedence over Altair's native `width` value.
+
+        Example
+        -------
+
+        >>> import pandas as pd
+        >>> import numpy as np
+        >>> import altair as alt
+        >>>
+        >>> df = pd.DataFrame(
+        ...     np.random.randn(200, 3),
+        ...     columns=['a', 'b', 'c'])
+        ...
+        >>> c = alt.Chart(df).mark_circle().encode(
+        ...     x='a', y='b', size='c', color='c', tooltip=['a', 'b', 'c'])
+        >>>
+        >>> st.beta_altair_chart(c, use_container_width=True)
+
+        Examples of Altair charts can be found at
+        https://altair-viz.github.io/gallery/.
+
+        """
+        proto = BetaVegaLiteChartProto()
+        marshall(
+            proto,
+            altair_chart,
+            use_container_width=use_container_width,
+        )
+
+        return self.dg._enqueue("beta_vega_lite_chart", proto)
+
+    @property
+    def dg(self) -> "streamlit.delta_generator.DeltaGenerator":
+        """Get our DeltaGenerator."""
+        return cast("streamlit.delta_generator.DeltaGenerator", self)
+
+
+def _is_date_column(df, name):
+    """True if the column with the given name stores datetime.date values.
+
+    This function just checks the first value in the given column, so
+    it's meaningful only for columns whose values all share the same type.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+    name : str
+        The column name
+
+    Returns
+    -------
+    bool
+
+    """
+    column = df[name]
+    if column.size == 0:
+        return False
+    return isinstance(column[0], date)
+
+
+def generate_chart(chart_type, data, width=0, height=0):
```

Comment:
Looks like `generate_chart` is only used by beta_altair.py - rename to `_generate_chart`?

Also needs a short docstring

Reply:
Addressed